icu4j/main/classes/core/src/com/ibm/icu/impl/units/UnitsRouter.java

```diff
@@ -47,16 +48,16 @@ public class UnitsRouter {
     private ArrayList<ConverterPreference> converterPreferences_ = new ArrayList<>();
 
     public UnitsRouter(String inputUnitIdentifier, String region, String usage) {
-        this(MeasureUnitImpl.forIdentifier(inputUnitIdentifier), region, usage);
+        this(MeasureUnitImpl.forIdentifier(inputUnitIdentifier), ULocale.forLanguageTag(region), usage);
```

Comment:
This will not work. forLanguageTag('US') will think it is a language called 'us'. You would need to use ULocale.forLanguageTag("und-" + region) or use the ULocale builder.

Reply:
Note, I suspect that longer term it would be better to push these up to the caller, eg make the call sites supply a locale.

Reply:
This an internal class that only used for exposing UnitsRouter as a class for the end users. Therefore, it is okay to leave this constructor taking the locale as a string. I used `new Ulocale(locale)`

Reply:
You still need to be consistent about whether the test uses a region subtag or a locale ID, and construct a Locale/ULocale accordingly. I added more verbose comments on the current version of the code, both in C++ and in Java.

Reply:
This code is slightly different now but still doesn't work. I am resolving this comment in favor of newer ones.